test/cpp/jit/CMakeLists.txt

```diff
@@ -17,6 +17,16 @@ add_library(backend_with_compiler SHARED
         )
 target_link_libraries(backend_with_compiler torch)
 
+# Don't link nnapi_backend library on MacOS
+if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
```

Comment:
This change should have been guarded with `AND BUILD_PYTHON`)

Reply:
Also, this is a bit weird place to keep build definitions for library that depends on libtorch_python